### PR TITLE
chore: update babel-presets, plugins and peerdeps

### PR DESCRIPTION
**Pull request by nikhedonia**
please have a close look.
I've added also babel-plugin-styled-components to ensure SSR classNames match client-side classNames.
This plugin causes all snapshots containing classnames to change.

"flow-react-proptypes" is not used in `ad` and `brightcove` as it breaks some tests.

### CODE REVIEW

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/0bf705de2187203f48f7f91c717c782606b8e91e/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/2bfe060626eb4ee8fb263fd3acee0b05d8fdd97b/index.html

File: package.json [hunk 1]
```diff
@@ -152,7 +152,8 @@
     "webpack": "3.3.0"
   },
   "dependencies": {
-    "babel-preset-env": "1.6.1",
-    "babel-preset-stage-2": "6.24.1"
+    "babel-plugin-add-react-displayname": "0.0.5",
```
Comment (craigbilner) [inline package.json]: does the "repo shell" need this?

File: packages/author-head/package.json [hunk 2]
```diff
@@ -78,8 +78,10 @@
     "svgs": "3.2.1"
   },
   "peerDependencies": {
-    "babel-preset-env": "1.6.1",
-    "babel-preset-stage-2": "6.24.1",
+    "babel-plugin-flow-react-proptypes": ">=14.0.3",
```
Comment (craigbilner) [inline packages/author-head/package.json]: exact types

File: packages/topics/package.json [hunk 3]
```diff
@@ -54,6 +54,16 @@
     "lodash.orderby": "4.6.0",
     "prop-types": "15.6.0"
   },
+  "peerDependencies": {
+    "babel-plugin-flow-react-proptypes": ">=14.0.3",
```
Comment (craigbilner) [inline packages/topics/package.json]: dev deps?

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/fd5538abca25ab7b597cf1f6987ad8ff7cc5579e/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/11c93795fdf7551133f4d29bdbf560981ed5de06/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/3ed585ca70dfcaebc66ee811c336b31d2b60b0d5/index.html

File: packages/eslint-config-thetimes/package.json [hunk 4]
```diff
@@ -30,9 +32,6 @@
     "eslint-plugin-react": "7.4.0",
     "prettier": "1.8.2"
   },
-  "peerDependencies": {
```
Comment (craigbilner) [inline packages/eslint-config-thetimes/package.json]: needs to be here

File: packages/eslint-config-thetimes/package.json [hunk 5]
```diff
@@ -22,6 +22,8 @@
   "homepage": "https://github.com/newsuk/times-components#readme",
   "dependencies": {
     "babel-eslint": "8.2.1",
+    "babel-preset-react-native": "4.0.0",
```
Comment (craigbilner) [inline packages/eslint-config-thetimes/package.json]: probably don't need this?

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/f060dfb0e2cd29867494e82519de1bb6c19820b4/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/c1a7af32949f21a09fe44c3bb88f988f0b077588/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/f386d047b18536fe7b15f7206c9b386099ec249e/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/f12199f1c17537eb7e682b20b9fba51b04635be3/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/3afabf640bb7d46cd2bc750b3021613f4b23f4d0/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/392d4b603374be701c559dbac6faf4e85e5a9178/index.html

**Comment by times-tools** [comment]
Please find visual snapshots of your changed components here: https://s3-eu-west-1.amazonaws.com/times-components-snaps/839c6522b042e67546e4ca5a92c877bd9e073fde/index.html

**Pull request merged**